cpp/include/raft/neighbors/ivf_pq_helpers.cuh

```diff
@@ -407,5 +575,78 @@ void erase_list(raft::resources const& res, index<IdxT>* index, uint32_t label)
   ivf_pq::detail::erase_list(res, index, label);
 }
 
+/**
+ * @brief Public helper API for computing the index's rotation matrix.
+ *
+ * @tparam IdxT
+ * @param[in] res
+ * @param[inout] index
+ * @param[in] force_random_rotation whether to apply a random rotation matrix on the input data. See
+ * raft::neighbors::ivf_pq::index_params for more details.
+ */
+template <typename IdxT>
+void make_rotation_matrix(raft::resources const& res,
+                          index<IdxT>* index,
+                          bool force_random_rotation)
+{
+  raft::neighbors::ivf_pq::detail::make_rotation_matrix(res,
+                                                        force_random_rotation,
+                                                        index->rot_dim(),
+                                                        index->dim(),
+                                                        index->rotation_matrix().data_handle());
+}
+
+/**
+ * @brief Public helper API for externally modifying the index's IVF centroids.
+ *
+ * @tparam IdxT
+ * @param[in] res
+ * @param[inout] index
+ * @param[in] cluster_centers the new cluster centers
+ */
+template <typename IdxT>
+void set_centers(raft::resources const& res,
+                 index<IdxT>* index,
+                 device_matrix_view<const float, uint32_t> cluster_centers)
+{
+  RAFT_EXPECTS(cluster_centers.extent(0) == index->n_lists(),
+               "Number of rows in cluster centers and IVF centers are different");
+  RAFT_EXPECTS(cluster_centers.extent(1) == index->dim(),
+               "Number of columns in cluster centers and index dim are different");
+  ivf_pq::detail::set_centers(res, index, cluster_centers.data_handle());
+}
+
+template <typename IdxT>
```

Comment:
needs docs and usage example. Please make sure to outline the expectations wrt the index's state.

Reply:
Done